src/go/k8s/internal/controller/redpanda/redpanda_controller.go

```diff
@@ -119,11 +122,41 @@ type RedpandaReconciler struct {
 // +kubebuilder:rbac:groups=core,namespace=default,resources=events,verbs=create;patch
 
 // SetupWithManager sets up the controller with the Manager.
-func (r *RedpandaReconciler) SetupWithManager(mgr ctrl.Manager) error {
+func (r *RedpandaReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager) error {
+	if err := registerPodClusterIndex(ctx, mgr); err != nil {
+		return err
+	}
+
+	helmManagedPodPredicate, err := predicate.LabelSelectorPredicate(
+		metav1.LabelSelector{
+			MatchExpressions: []metav1.LabelSelectorRequirement{{
+				Key:      "app.kubernetes.io/name", // look for only redpanda or console pods
+				Operator: metav1.LabelSelectorOpIn,
+				Values:   []string{"redpanda", "console"},
+			}, {
+				Key:      "app.kubernetes.io/instance", // make sure we have a cluster name
+				Operator: metav1.LabelSelectorOpExists,
+			}, {
+				Key:      "batch.kubernetes.io/job-name", // filter out the job pods since they also have name=redpanda
+				Operator: metav1.LabelSelectorOpDoesNotExist,
+			}},
+		},
+	)
+	if err != nil {
+		return err
+	}
+
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&v1alpha2.Redpanda{}).
+		Owns(&sourcev1.HelmRepository{}).
 		Owns(&helmv2beta1.HelmRelease{}).
 		Owns(&helmv2beta2.HelmRelease{}).
+		Watches(&corev1.Pod{}, handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, o client.Object) []reconcile.Request {
```

Comment:
Did you consider instead watching the Statefulset/Deployments? They're already performing all these watches and calculations and we could just be passing along information from their statuses instead of skipping over them.

Reply:
I can drop the pod watch if you want. I actually just had to add a watch for the `StatefulSet` anyway due to the rolling restarts not being captured by solely looking at pod health. WDYT, drop the pod watch or just do both?

Reply:
I'm a big fan of separation of concerns. Let the StatefulSet deal with Pods and we'll deal with the StatefulSet?